src/Field.tsx

```diff
@@ -64,17 +64,33 @@ export interface FieldConfig {
   innerRef?: (instance: any) => void;
 }
 
-export type FieldAttributes<T> = GenericFieldHTMLAttributes & FieldConfig & T;
+export type FieldAttributes<T> = GenericFieldHTMLAttributes &
+  FieldConfig &
+  T & { name: string };
 
-export function useField<Val = any>(name: string, type?: string) {
+export function useField<Val = any>(
+  propsOrFieldName: string | FieldAttributes<Val>
+) {
   const formik = useFormikContext();
+
   if (process.env.NODE_ENV !== 'production') {
     invariant(
       formik,
       'useField() / <Field /> must be used underneath a <Formik> component or withFormik() higher order component'
     );
   }
-  return formik.getFieldProps<Val>(name, type);
+
+  if (isObject(propsOrFieldName)) {
+    if (process.env.NODE_ENV !== 'production') {
```

Comment:
minor, but I like more __DEV__ constant provided by tsdx (and I know you know)

Reply:
Let's make a different issue for that.

Reply:
Sure, I can do a find and replace on whole repo